csharp/native/SubTasking/Worker/SubTaskingWorker.cs

```diff
@@ -0,0 +1,261 @@
+// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using ArmoniK.Api.Common.Channel.Utils;
+using ArmoniK.Api.Common.Utils;
+using ArmoniK.Api.gRPC.V1;
+using ArmoniK.Api.gRPC.V1.Agent;
+using ArmoniK.Api.Worker.Worker;
+
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+
+using Microsoft.Extensions.Logging;
+
+using Empty = ArmoniK.Api.gRPC.V1.Empty;
+
+namespace ArmoniK.Samples.SubTasking.Worker
+{
+  public class SubTaskingWorker : WorkerStreamWrapper
+  {
+    /// <summary>
+    ///   Initializes an instance of <see cref="SubTaskingWorker" />
+    /// </summary>
+    /// <param name="loggerFactory">Factory to create loggers</param>
+    /// <param name="provider">gRPC channel provider to send tasks and results to ArmoniK Scheduler</param>
+    [Obsolete("Obsolete")]
+    public SubTaskingWorker(ILoggerFactory      loggerFactory,
+                            GrpcChannelProvider provider)
+      : base(loggerFactory,
+             provider)
+      => logger_ = loggerFactory.CreateLogger<SubTaskingWorker>();
+
+    /// <summary>
+    ///   Function that represents the processing of a task.
+    /// </summary>
+    /// <param name="taskHandler">Handler that holds the payload, the task metadata and helpers to submit tasks and results</param>
+    /// <returns>
+    ///   An <see cref="Output" /> representing the status of the current task. This is the final step of the task.
+    /// </returns>
+    public override async Task<Output> Process(ITaskHandler taskHandler)
```

Comment:
You can override ProcessAsync that takes a cancellation token instead. That's the new way to do it.